firedrake/variational_solver.py

```diff
@@ -366,7 +366,7 @@ def __init__(self, a, L, u, bcs=None, aP=None,
         else:
             if not isinstance(L, (ufl.BaseForm, slate.slate.TensorBase)):
                 raise TypeError("Provided RHS is a '%s', not a Form or Slate Tensor" % type(L).__name__)
-            if len(L.arguments()) != 1:
+            if len(L.arguments()) > 1:
```

Comment:
```suggestion
            if len(L.arguments()) != 1 and not L.empty():
```
Otherwise anything of rank 0 would be allowed.